site/src/modules/management/ManagementSettingsLayout.tsx

```diff
@@ -48,61 +47,128 @@ export const canEditOrganization = (
 	);
 };
 
+const isManagementRoutePermitted = (
```

Comment:
Long function, but I really wanted to make sure that users couldn't access pages that they shouldn't be able to

The previous logic was set up so that if a user could view the deployments page at all, they would be able to update the URL and jump to any page. Felt like we needed something more strict than just hiding navigation options in the sidebar

Reply:
Maybe extract this into an util and add specific tests to it?

Reply:
Yeah, I can do that really quickly